### DISCUSSION TITLE
3.6.1 changes shower time, howto from HA?

### DISCUSSION THREAD

**Discussion by kroon040** [emsesp/EMS-ESP32#1293] [Q&A] [answered]
Hi,

I see that there is a breaking change for de shower duration in minutes to report.

I have this before 3.6.1 in HA. I didn't find already a solution in the wiki

`alias: Alert shower time
trigger:
  - platform: state
    entity_id: sensor.last_shower_duration
action:
  - service: notify.mobile_app_ikroonen
    data_template:
      message: "Duration: {{ states('sensor.last_shower_duration') }}"
      title: Shower finished at {{ now().strftime("%H:%M") }}
initial_state: true`

**Comment by proddy** [comment]
yes it changed, and the wiki was updated [here](https://emsesp.github.io/docs/Home-Assistant/#shower-integration) with new examples, however I just noticed a few formatting issues so will fix those later.

I also want to simplify it and use the latest template sensors in HA (https://www.home-assistant.io/blog/2023/09/06/release-20239/#template-sensors-from-the-ui)

**Comment by kroon040** [comment]
But I see the time druation notify is in seconds, it shpul be nice to have it in minutes and seconds again

**Reply by proddy** [reply]
ok, the reason being?

HA uses seconds as the duration. I can make it a HH:MM sensor string like last time, but it won't be following the standards

**Comment by kroon040** [comment]
I don't think it should be change that you send from ems-esp32, but I think the automation must change for nice formatting in MM:SS

calculating it there

**Reply by proddy** [reply]
that is how it works, when HA displays the time it will format it, like MM:SS or any other format the user wants. This is easy to do with a template sensor. I will update the docs to reflect this

**Comment by kroon040** [comment]
somehing like this post?

https://community.home-assistant.io/t/solved-convert-seconds-in-mm-ss/446476

**Reply by proddy** [reply]
That would work as well. The wiki
Already shows a full working example of converting to MM:SS which is simpler.

**Comment by kroon040** [comment]
I don't see him yet in the wiki, last is still form 29 august.

**Reply by proddy** [reply]
it's there in the link I provided earlier. I added a few more spaces so it's readable.

**Comment by NongJoWo** [comment]
Hi, is there any possibility in ESP config to set the duration trigger of a shower higher than 2min? Many thanks in advance...

**Reply by NongJoWo** [reply]
I am looking for this setting: SHOWER_MIN_DURATION - xx minutes, before recognizing its a shower

**Reply by proddy** [reply]
yes, we can easily make that variable and a setting in the WebUI if that is what you need. Or you can fork this repo and change the line in `shower.h`

**Reply by NongJoWo** [reply]
If we can make that variable and a setting in the WebUI would be great.

**Reply by proddy** [reply, marked as answer]
#1801
